### DISCUSSION TITLE
Autodiscover filter to include whole group, except for selected projects

### DISCUSSION THREAD

**Discussion by szpak** [renovatebot/renovate#21341] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.129.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'm introducing Renovate in my organization (using self-hosted GitLab) . I started by adding some pilot project one by one and it worked fine. As the second step, I would like to be able to add the whole GitLab groups, one by one (to have a look at the performance metrics). It can be easily achieved with the autodiscovery and adding a Renovate user to the whole group. However, in those groups there are also some technical/deprecated projects and I would prefer to have them ignored by Renovate (to not generate fuzz and to do not unnecessary spend time on them).

### What you tried so far.

I wanted to use the autodiscovery filer (in the global configuration) as follows, but it didn't work:

```
  autodiscover: true,
  autodiscoverFilter: [
    "whole-group1/*",
    "!/whole-group1/legacy-project-to-skip/",
    ...
```

It might be causes by the fact that OR is used on the filters.

Is it possible to add the whole group, but exclude some of its projects? I don't want to add all the project in a group by hand to automatically detect new projects.

### Relevant debug logs

<details><summary>Logs</summary>

```
INFO: Autodiscovered repositories
       "length": 10,
       "repositories": [
         "whole-group1/wanted-project",
         "whole-group1/legacy-project-to-skip",
...
```

</details>

**Comment by rarkins** [comment, marked as answer]
I confirmed in the docs, it says "If you set multiple filters, then the matches of each filter are added to the overall result."

i.e. it's OR functionality, not AND.

@secustor wrote this and may have ideas

**Reply by secustor** [reply]
No, there is currently no way to this in the discovery phase. 

What you can do is 
1. Archive the project, because Renovate will ignore archived projects during discovery 
https://github.com/renovatebot/renovate/blob/1fede64520394596a01d17466a5dc3eac9156106/lib/modules/platform/gitlab/index.ts#L146
2. Add such a `renovate.json` to your project. 
```json
{
  "enabled": false
}
```

**Reply by rarkins** [reply]
3. Create a feature request issue for "autodiscoverExclusions", which will be patterns to exclude, and they will be ANDed together. So we would OR together autodiscoverFilter and then AND together all in autodiscoverExclusions

**Reply by szpak** [reply]
Thanks for the confirmation it is currently problematic to achieve.

I use onboarding, so closing the initial onboarding MR in those project should tell Renovate to ignore them, so it is also some approach (similar to `enabled: false`).

Nevertheless, it could be more clear to have a way to exclude to projects/subgroups in the global configuration, so I created a future request: https://github.com/renovatebot/renovate/issues/21347
